### DISCUSSION TITLE
[Bug]Issues Setting up Harvester - SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED

### DISCUSSION THREAD

**Discussion by ChiaFarmer86** [Chia-Network/chia-blockchain#10167] [Support]
### What happened?

Hello,

I am in desperate need of help. I've been trying to setup a harvester in the same location (network) as my farmer. I've followed all of the steps on how to get the process done and no matter what I do, I keep getting the following error message in the Harvesters Log:

2022-02-05T19:48:20.757 harvester harvester : INFO Reconnecting to peer {'host': '10.0.1.99', 'port': 8447}
2022-02-05T19:48:31.601 harvester harvester_server : INFO Cannot connect to host 10.0.1.99:8447 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate signature failure (_ssl.c:1129)')]

I've followed each of the steps here: https://github.com/Chia-Network/chia-blockchain/wiki/Farming-on-many-machines

I tried running telnet 10.0.1.99 to make sure I can connect to my farmer and the output I got was:

Trying 10.0.1.99...
Connected to 10.0.1.99.
Escape character is '^]'.
Connection closed by foreign host.

Both machines are running Ubuntu 21.10 and my router is an Apple Airport Extreme.

I would be SUPER APPRECIATIVE if someone can please help me.

### Version

1.2.11

### What platform are you using?

Linux

### What ui mode are you using?

GUI

### Relevant log output

```shell
2022-02-05T19:48:20.757 harvester harvester : INFO Reconnecting to peer {'host': '10.0.1.99', 'port': 8447}
2022-02-05T19:48:31.601 harvester harvester_server : INFO Cannot connect to host 10.0.1.99:8447 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: certificate signature failure (_ssl.c:1129)')]
```

**Comment by stefan-lange** [comment]
Should be the wrong ca

1. Go to the farmer ~/.chia/mainnet/config/ssl/ca
2. Copy this ca folder to the harvester e.g. /tmp/farmer-ca
3. Run on harvester `chia init -c /tmp/farmer-ca`
4. Start the harvester

**Comment by ChiaFarmer86** [comment]
Thank you for trying to help, but unfortunately it's not that simple. I did that process and thats not the issue. I tried that process multiple times, even wiped the entire computer clean, did a fresh install of CLI, then moved the CA folder from the farmer to the harvester, ran the chia init command and started the harvester, no luck. 

I think my issue is with either my router or some sort of settings from within the harvester that is closing out the connection. If you see the output I get when I run telnet you might realize what I mean. I am using an Apple Airport Extreme, maybe this is the issue? Maybe I have to do something within the settings. 

I've been struggling with this for weeks, cant figure it out.

**Comment by stefan-lange** [comment]
Hm, ok, that's sad. 2 days ago I had exactly the same issue (certificate signature failure) with one of my harvesters within docker swarm. Used the chia_keys format everywhere, but that wasn't working. Connection from harvester to farmer was ok.

You could check the 'standard' stuff. Firewalls, configured ports (farmer port 8447) etc.

**Reply by ChiaFarmer86** [reply]
Im sure someone else must have had my same issue, hopefully they are willing to help. 

- Farmer Port is open. 
- I am copying the CA folder from the farmer to the harvester
- Firewall on Ubuntu is inactive
- Correct IP address (farmer) is being inputted into the config file on the harvester 

I am 99% sure the issue is not with the chia software. It has to be something with my router, switch or computers? I can't figure it out. Like I mentioned before, when I ran telnet, it shows connected, then disconnects a couple seconds later. 

Hope someone will eventually help. Im willing to pay someone if they can spend the time to trouble and help me. 

Thank you,

**Reply by stefan-lange** [reply]
Did you copy the farmers `ca` folder to `~/.chia/mainnet/config/ssl/ca` on the harvester? If so, it won't work. The `chia init -c` must be executed in order to generate new and correct certificates for the harvester. Otherwise the 'certificate signature failure' from you log will be shown.

**Reply by ChiaFarmer86** [reply]
I copied the CA folder from the farmer to the desktop of the harvester. Then used the chia init - c command to execute it. All of the prompts I received after doing that command were correct, so I know it went through (I was following videos online of other people doing it to, and everything was on point). 

But once everything is done, I check my farmer and I don't see the harvester listed as a connection, I don't see the total number of plots changing :(. 

IF running telnet and it shows the connection disconnecting, then its definitely not a chia issue right? Its something with the network? 

Maybe I shouldn't be using a Apple Airport for my router OR maybe my switch is blocking the connection????

Unfortunately I am running a 2nd full node now till I get this fixed. Strange why Chia doesn't have tech support guys trying to help. We're investing tons of money into the project and there are so many discussions and bug issues that are completely unanswered. 

Im hopeful with more troubleshooting this will be resolved one day, hahahah!

Thank you Stefan for your help though, really appreciate it.

**Comment by loppefaaret** [comment]
when you do the telnet test - how long is the connection open before it gets closed by the remote host ? do you see anything on the remote host system logs ?

**Comment by spik3yp** [comment]
for future readers. Please follow these steps:
- copy the "ca" folder from the full node machine, usually located at C:\Users\{local user name}\.chia\mainnet\config\ssl
- paste the "ca" folder to the harvester C:\ 
- open up command prompt or Powershell
- type in: chia init -c C:\ca

you should see this line in the output text:
"Found private CA in C:\Users\{local user name}\.chia\mainnet, using it to generate TLS certificates"
It's all set and good now.
If you don't see it, then you have to make sure that your "chia init -c" path is including the "ca" folder in the path, not just the path to the parent folder of "ca" folder.

**Comment by Jufik** [comment]
Had similar issues running on Ubuntu.
Wiped the whole `mainnet` folder and ran `chia init -c /tmp/ca` again: it worked :) 

No `CERTIFICATE_VERIFY_FAILED` anymore, even though when firing up the harvester, all I get in log is:
`daemon chia.daemon.server         : DEBUG    About to ping: chia_harvester`

Looks like nothing happens between both nodes, found nothing related to the worker node in main node logs......

**Reply by BrandtH22** [reply]
Hey, is the main node able to detect the plots on the harvester?

**Reply by Jufik** [reply]
No, nothing happens. I do not think that's a Firewall issue, instances can communicate with each other.
I'll just fire up multiple nodes ATM

**Reply by BrandtH22** [reply]
Hey, as a heads up we can generally provide more timely and thorough support on our discord server (https://discord.gg/chia).

I would go back through this doc and make sure everything is setup properly then verify that the farmer can see the plots from the harvester:
https://docs.chia.net/farming-on-many-machines/

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
